// Copyright (C) 2019 by Jakub Wojciech

// This file is part of Lelo Remote Music Player.

// Lelo Remote Music Player is free software: you can redistribute it
// and/or modify it under the terms of the GNU General Public License
// as published by the Free Software Foundation, either version 3 of
// the License, or (at your option) any later version.

// Lelo Remote Music Player is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty
// of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
// General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Lelo Remote Music Player. If not, see
// <https://www.gnu.org/licenses/>.

syntax = "proto3";

message CertClientMessage {
  message CertRequest {
    bytes request = 1;
  }
  message ConfirmRequest {
    bytes cert_hash = 1;
  }

  oneof Content {
    CertRequest cert_request = 1;
    ConfirmRequest confirm_request = 2;
  }
}

message CertServerMessage {
  message ConfirmResponse {
    bool response = 1;
  }
  message CertBundle {
    bytes client_cert = 1;
    bytes root_cert = 2;
  }

  oneof Content {
    int32 error_code = 1;
    ConfirmResponse confirm_response = 2;
    CertBundle cert_bundle = 3;
  }
}